crates/interpreter/src/valid.rs

```diff
@@ -412,24 +406,32 @@ struct Expr<'a, 'm> {
     is_body: bool,
     locals: Vec<ValType>,
     labels: Vec<Label<'m>>,
+    side_table: Vec<Option<SideTableEntryView>>,
+}
+
+#[derive(Debug, Default, Copy, Clone, PartialEq, Eq)]
+struct SideTableBranch<'m> {
+    parser: &'m [u8],
+    side_table_len: usize,
 }
 
 #[derive(Debug, Default)]
 struct Label<'m> {
     type_: FuncType<'m>,
     /// Whether an `else` is possible before `end`.
-    kind: LabelKind,
+    kind: LabelKind<'m>,
     /// Whether the bottom of the stack is polymorphic.
     polymorphic: bool,
     stack: Vec<OpdType>,
+    branches: Vec<SideTableBranch<'m>>,
 }
 
 #[derive(Debug, Default, Copy, Clone, PartialEq, Eq)]
```

Comment:
Actually, now that we're making this enum more than just an integer, we should remove `Copy, PartialEq, Eq`. This was just used for `If` as convenience. But we should use `match` and `matches!` now.